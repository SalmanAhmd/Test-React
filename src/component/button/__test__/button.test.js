import React from 'react';
import ReactDom from 'react-dom';

import Button from '../';

import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';


it('render button without crash', () => {
    const div = document.createElement('div');
    ReactDom.render(<Button />, div)
})

it('render button correctly', () => {
    const { getByTestId } = render(<Button label='button label' />);
    expect(getByTestId('button')).toHaveTextContent('button label')
})

it('render button correctly 1', () => {
    const { getByTestId } = render(<Button label='Save' />);
    expect(getByTestId('button')).toHaveTextContent('Save')
})

it('matches button snapshot', () => {
    const tree = renderer.create(<Button label='button label' />).toJSON();
    expect(tree).toMatchSnapshot();
})

it('matches button snapshot 2', () => {
    const tree = renderer.create(<Button label='button label 2' />).toJSON();
    expect(tree).toMatchSnapshot();
})
