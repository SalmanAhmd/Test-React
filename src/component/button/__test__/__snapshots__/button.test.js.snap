// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches button snapshot 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  Hi 
  button label
</div>
`;

exports[`matches button snapshot 2 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  Hi 
  button label 2
</div>
`;
